net/openfortivpn/files/openfortivpn.sh

```diff
@@ -0,0 +1,139 @@
+        proto_config_add_string "username"
+        proto_config_add_string "password"
+        proto_config_add_string "trusted_cert"
+        proto_config_add_int "set_dns"
+        proto_config_add_int "pppd_use_peerdns"
+        proto_config_add_int "metric"
+        no_device=1
+        available=1
+}
+
+proto_openfortivpn_setup() {
+        local config="$1"
+
+        json_get_vars host server iface_name local_ip username password trusted_cert set_dns pppd_use_peerdns metric
+
+        ifname="vpn-$config"
+
+        logger -t openfortivpn "$config: initializing..."
+
+        [ -n "$iface_name" ] && {
+            json_load "$(ifstatus $iface_name)"
+            json_get_var iface_device_name device
+            json_get_var iface_device_up up
+        }
+
+        logger -t "openfortivpn" "$config: $iface_name is status  $iface_device_up"
+        [ "$iface_device_up" -eq 1 ] || {
+            logger -t "openfortivpn" "$config: $iface_name is not up $iface_device_up"
+            proto_notify_error "$config" "$iface_name is not up $iface_device_up"
+            proto_block_restart "$config"
+            exit 1
+        }
+
+
+        server_ip=$(resolveip -t 10 "$server")
+
+        [ $? -eq 0 ] || {
+            logger -t "openfortivpn" "$config: failed to resolve server ip"
+            sleep 10
+            proto_notify_error "$config" "failed to resolve server ip"
+            proto_setup_failed "$config"
+            exit 1
+        }
+
+        [ -n $iface_name ] && {
+            ping -I $iface_device_name -c 1 -w 10 $server_ip > /dev/null 2>&1 || {
+                logger -t "openfortivpn" "$config: failed to ping $server_ip on $iface_device_name"
+                sleep 10
+                proto_notify_error  "$config" "failed to ping $server_ip on $iface_device_name"
+                proto_setup_failed "$config"
+                exit 1
+            }
+        }
+
+        for ip in $(resolveip -t 10 "$server"); do
+                logger -t "openfortivpn" "$config: adding host dependency for $ip on $iface_name at $config"
+                proto_add_host_dependency "$config" "$ip" "$iface_name"
+        done
+
+
+
+        [ -n "$port" ] && port=":$port"
+
+        append_args "$server$port" --pppd-ifname="$ifname" --use-syslog  -c /dev/null
+        append_args "--set-dns=$set_dns"
+        append_args "--no-routes"
+        append_args "--pppd-use-peerdns=$pppd_use_peerdns"
+
+        [ -n "$iface_name" ] && {
+            append_args "--ifname=$iface_device_name"
+        }
+
+        [ -n "$trusted_cert" ] && append_args "--trusted-cert=$trusted_cert"
+        [ -n "$username" ] && append_args -u "$username"
+        [ -n "$password" ] && {
+                umask 077
+                mkdir -p /var/etc
+                pwfile="/var/etc/openfortivpn-$config.passwd"
+                echo "$password" > "$pwfile"
+        }
+
+        [ -n "$local_ip" ] || local_ip=192.0.2.1
+        mkdir -p '/etc/ppp/peers'
```

Comment:
I think this could be also on /var so no flash writes

Reply:
Will openwrt `pppd` read from /var? I know on other installations, it is fixed to read from /etc/ppp/peers for security reasons.  Would be nice to avoid the flash writes though.